common/src/main/java/org/wildfly/httpclient/common/Version.java

```diff
@@ -20,12 +20,18 @@
 /*
  * Versioning enum for HttpHandler implementations.
  *
+ * TODO: due to the way EENamespaceInteroperability.createInteroperabilityHandler(HttpHandler...) works,
+ * the original protocol versions JAVAEE_PROTOCOL_VERSION and JAKARTA_PROTOCOL_VERSION need to share
+ * the same handler instance, so it was not possible to match protocol handler indexes to protocol versions
+ * in a 1-to-1 manner. In order to avoid a confusing protocol version to handler version mismatch, they share
+ * the handler installed at index 2.
  * TODO: integrate this with the Protocol class in a nice way.
```

Comment:
How can this be integrated if  the idea was to decouple the protocol version from the handlers version?
Maybe we should rename this Version class to HandlerVersion to make it clearer?
And how does this interplay with incremented handler versions in the future, supposing only one type of handler needs to have its version incremented? Say, for example, that only Transaction handlers will need a new request header, or a new parameter, or anything really that is valid only for transaction handlers.

Reply:
I have renamed the class. These are good questions. And what if there are Protocol version changes which don't correspond to handler version changes.

Reply:
I just lost a well-crafted 200 word response to this with one misplaced click which killed my text :-( 
I am looking at this again and here is my idea in a nutshell:
1. introduce one Handler class for each of org.wildfly.httpclient.{ejb, transaction, naming} which will contain the handler version for that module, just as the Protocol class holds the protocol version to be used in org.wildfly.httpclient.common. Handler.LATEST serves the same purpose. Use this version when building client requests for ejb, transaction and naming invocations. In other words, the version in the context path is the handler version.
2. In the handshake between ClientConnectionHolder, the server-side wrapper functionality of EENamespaceInteroperabilityHandler / JakartaNamespaceHandler and EEIneroperableClientExchange, set things up so that only one combined wrapper is used on the server side and it is based only on reading the x-wf-version HTTP header to decide which marshaller and unmarshaller factories to attach to requests and reploes.